File: composeApp/src/androidMain/kotlin/com/muralex/myapp/screens/home/HomeScreenEvents.kt
package com.muralex.myapp.screens.home

import com.muralex.myapp.navigation.DetailsNavParams
import com.muralex.myapp.navigation.ScreenNavActions
import com.muralex.myapp.navigation.SectionNavParams
import com.muralex.myapp.viewmodel.screens.home.HomeListItem
import com.muralex.myapp.viewmodel.screens.home.HomeSectionState

sealed class HomeUiEvent {
    data class OnItemClicked(val item: HomeListItem) : HomeUiEvent()
    data class OnSectionClicked(val section: HomeSectionState) : HomeUiEvent()
}

class HomeEventHandler(
    val navActions: ScreenNavActions
) {
    fun onEvent(event: HomeUiEvent) {
        when (event) {
            is HomeUiEvent.OnItemClicked ->
                navActions.toDetailFromList(event.item.toDetailsNavParams())
            is HomeUiEvent.OnSectionClicked ->
                navActions.toSection(event.section.toSectionNavParams())
        }
    }
}

fun HomeListItem.toDetailsNavParams() = DetailsNavParams(id, name)
fun HomeSectionState.toSectionNavParams() = SectionNavParams(sectionId, sectionName)

File: iosApp/iosApp/views/screens/home/HomeScreenEvents.swift

import SwiftUI
import Shared


enum HomeUiEvent {
    case onItemClicked(HomeListItem)
    case onSectionClicked(HomeSectionState)
}

struct HomeEventHandler {
    let navActions: NavigationActions
    
    func onEvent(_ event: HomeUiEvent) {
        switch event {
        case .onItemClicked(let item):
            navActions.toDetailFromList(item.toDetailsNavParams())
        case .onSectionClicked(let section):
            navActions.toSection(section.toSectionNavParams())
        }
    }
}

extension HomeListItem {
    func toDetailsNavParams() -> DetailsNavParams {
        DetailsNavParams(id: id, name: name)
    }
}

extension HomeSectionState {
    func toSectionNavParams() -> SectionNavParams {
        SectionNavParams(id: sectionId, name: sectionName)
    }
}
